services/search/pkg/engine/bleve.go

```diff
@@ -206,7 +206,7 @@ func (b *Bleve) Search(_ context.Context, sir *searchService.SearchIndexRequest)
 
 	return &searchService.SearchIndexResponse{
 		Matches:      matches,
-		TotalMatches: int32(res.Total),
+		TotalMatches: int32(len(matches)),
```

Comment:
I don't think this is the proper fix. If the search is paginated by n you will only get n results in `matches` even though the total number of matches can be much higher.

What we could maybe do instead is decreasing the number of total matches when disregarding matches in https://github.com/owncloud/ocis/blob/master/services/search/pkg/engine/bleve.go#L167 - which I believe is the source of the # of total matches mismatch.